rootex/core/renderer/materials/textured_material.cpp

```diff
@@ -38,3 +57,57 @@ void TexturedMaterial::bind()
 	setPSConstantBuffer({ LightSystem::GetSingleton()->getLights() });
 	setPSConstantBuffer(PSDiffuseConstantBufferMaterial({ 0.6f, 30.0f, { 0.0f, 0.0f } }));
 }
+
+JSON::json TexturedMaterial::getJSON() const
+{
+	JSON::json j;
+
+	j["type"] = "Textured Material";
+	j["imageFile"] = m_ImageFile->getPath();
+
+	return j;
+}
+
+void TexturedMaterial::setTexture(ImageResourceFile* image)
+{
+	Ref<Texture> texture(new Texture(image));
+	m_ImageFile = image;
+	m_DiffuseTexture = texture;
+}
+
+#ifdef ROOTEX_EDITOR
+#include "imgui.h"
+#include "imgui_stdlib.h"
+void TexturedMaterial::draw()
+{
+	ImGui::Text("Textured Material");
+	m_ImagePathUI = m_ImageFile->getPath().string();
```

Comment:
I think this line would never let you edit the path from the ImGui::InputText box. Check if this works

Reply:
This works, `setTexture(image)` changes `m_ImageFile`